packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2355,18 +2371,32 @@ class RenderTransform extends RenderProxyBox {
   void paint(PaintingContext context, Offset offset) {
     if (child != null) {
       final Matrix4 transform = _effectiveTransform!;
-      final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
-      if (childOffset == null) {
-        layer = context.pushTransform(
-          needsCompositing,
-          offset,
-          transform,
-          super.paint,
-          oldLayer: layer as TransformLayer?,
-        );
+      if (filterQuality == null) {
+        final Offset? childOffset = MatrixUtils.getAsTranslation(transform);
+        if (childOffset == null) {
+          layer = context.pushTransform(
+            needsCompositing,
+            offset,
+            transform,
+            super.paint,
+            oldLayer: layer is TransformLayer ? layer as TransformLayer? : null,
+          );
+        } else {
+          super.paint(context, offset + childOffset);
+          layer = null;
+        }
       } else {
-        super.paint(context, offset + childOffset);
-        layer = null;
```

Comment:
We still need to null out the layer if child is null, no?

Reply:
We don't need to. It won't be used on this pass, but it might be used if we have a new frame that does have a child. It wasn't originally cleared in the null child case, so that hasn't changed here.

Reply:
I've still left this in the latest push because it was present in existing logic so I didn't want to change code I wasn't specifically introducing to fix the issue.